File: 28.实现 strStr()_暴力解法.cc
#include <string>
using std::string;

class Solution {
public:
    int strStr(string haystack, string needle) {
        for (int i = 0; i + needle.size() <= haystack.size(); ++i) {
            bool match = true;
            for (int j = 0; j < needle.size(); ++j) {
                if (i + j == haystack.size()) {
                    return -1;
                }
                if (haystack[i + j] != needle[j]) {
                    match = false;
                    break;
                }
            }
            if (match) {
                return i;
            }
        }
        return -1;
    }
};

File: Rust/28.实现 strStr() 暴力解法.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        if needle.len() == 0 {
            return 0;
        }
        for i in 0..haystack.len() - needle.len() + 1 {
            let mut matched: bool = true;
            for j in 0..needle.len() {
                if i + j > haystack.len() {
                    return -1;
                }
                if haystack.chars().nth(i + j) != needle.chars().nth(j) {
                    matched = false;
                    break;
                }
            }
            if matched {
                return i as i32;
            }
        }
        return -1;
    }
}
